### ISSUE TITLE
`F821` false-positive with class member aliases

### ISSUE THREAD

**Issue by Avasam** [PyCQA/flake8-pyi#317]
If I'm not mistaken flake8-pyi does some things for pyflakes to act correctly in a stub file, so hopefully the issue is opened in the right place.
I was working on some stubs today for a library that aliases class members. Instead of have to duplicate some definitions multiple times, I went with the following (minimal repro, there was obviously a lot more aliases and complex types)

`.pyi` file:
```python
class MyClass:
    foo: int
    bar = foo  # undefined name 'foo'Flake8(F821)

reveal_type(MyClass().bar)  # note: Revealed type is "builtins.int"
```

Workarounds: `#noqa: F821`, `foo: int = ...` or `bar: int` (but I kinda liked the obvious semantic saying it's an alias)

**Comment by AlexWaygood** [comment]
> flake8-pyi does some things for pyflakes to act correctly in a stub file

Yeah we do some fairly ghastly monkey-patching to workaround the fact that pyflakes doesn't support `.pyi` files (and, though we've never directly asked them, I doubt they'd be interested in accepting patches so that they _did_ support `.pyi` files).

I kind of don't like the idea of touching the bit of the code where we do the monkey-patching, but maybe it's time that we bit the bullet and tried to improve it ☹️

**Comment by Avasam** [comment]
In a project that also uses typecheckers that can warn about undefined types (ie: typeshed 😉  ). Would it be sensible to disable F821 for `.pyi` files?

Recommending to disable the check might be less complicated and pushes back the need to touch that area of code to aovid false-positives.

**Comment by AlexWaygood** [comment]
Perhaps. I shall think on this and experiment a little :)
